assignments/prototypes.js

```diff
@@ -7,22 +7,47 @@
   
   Each constructor function has unique properites and methods that are defined in their block comments below:
 */
-  
-/*
+
+function GameObject(createdAt, dimensions){
+  this.createdAt = createdAt;
+  this.dimensions = dimensions;
+}
+GameObject.prototype.destroy = function(){
+  return `${this.name} was removed from the game.`;
+}
+  /*
   === GameObject ===
   * createdAt
   * dimensions
   * destroy() // prototype method -> returns the string: 'Object was removed from the game.'
 */
-
+function CharacterStats(createdAt, dimensions, hp, name){
+  GameObject.call(this, createdAt, dimensions);
+  this.hp = hp;
+  this.name =name;
+}
+CharacterStats.prototype = new GameObject();
+CharacterStats.prototype.takeDamage = function(){
+  return `${this.name} took damage.`
+}
 /*
   === CharacterStats ===
   * hp
   * name
   * takeDamage() // prototype method -> returns the string '<object name> took damage.'
   * should inherit destroy() from GameObject's prototype
 */
-
+function Humanoid({createdAt, dimensions, hp, name, faction, weapons, language}){
```

Comment:
[in case you want to know more about destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment)